File: app/src/entourage/java/social/entourage/android/MainPresenter.java
package social.entourage.android;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Intent;
import android.net.Uri;
import android.widget.Toast;

import com.google.firebase.iid.FirebaseInstanceId;

import javax.inject.Inject;

import social.entourage.android.about.EntourageAboutFragment;
import social.entourage.android.api.AppRequest;
import social.entourage.android.api.UserRequest;

import static android.content.Context.CLIPBOARD_SERVICE;

/**
 * Presenter controlling the MainActivity
 * @see MainActivity
 */
public class MainPresenter extends MainBasePresenter {

    // ----------------------------------
    // LIFECYCLE
    // ----------------------------------
    @Inject
    MainPresenter(final MainActivity activity, final AppRequest appRequest, final UserRequest userRequest) {
        super(activity, appRequest, userRequest);
    }

    // ----------------------------------
    // PRIVATE METHODS
    // ----------------------------------
    @Override
    protected void handleMenu(final int menuId) {
        if (activity == null) return;
        switch (menuId) {
            case R.id.action_ambassador:
                EntourageEvents.logEvent(EntourageEvents.EVENT_MENU_AMBASSADOR);
                Intent ambassadorIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(activity.getLink(Constants.AMBASSADOR_ID)));
                try {
                    activity.startActivity(ambassadorIntent);
                } catch (Exception ex) {
                    Toast.makeText(activity, R.string.no_browser_error, Toast.LENGTH_SHORT).show();
                }
                break;
            case R.id.mainprofile_app_version:
                final android.content.ClipboardManager clipboardManager = (ClipboardManager)EntourageApplication.get().getSystemService(CLIPBOARD_SERVICE);
                if(clipboardManager!=null){
                    ClipData clipData = ClipData.newPlainText("FirebaseID", FirebaseInstanceId.getInstance().getId());
                    clipboardManager.setPrimaryClip(clipData);
                }
                break;
            case R.id.action_about:
                EntourageEvents.logEvent(EntourageEvents.EVENT_MENU_ABOUT);
                EntourageAboutFragment aboutFragment = new EntourageAboutFragment();
                aboutFragment.show(activity.getSupportFragmentManager(), EntourageAboutFragment.TAG);
                break;
            default:
                super.handleMenu(menuId);
                break;
        }
    }
}

File: app/src/pfp/java/social/entourage/android/MainPresenter.kt
package social.entourage.android

import android.content.Intent
import android.net.Uri
import android.widget.Toast
import androidx.annotation.IdRes
import social.entourage.android.about.VoisinageAboutFragment
import social.entourage.android.api.AppRequest
import social.entourage.android.api.UserRequest
import social.entourage.android.privateCircle.PrivateCircleChooseFragment
import javax.inject.Inject

/**
 * Presenter controlling the MainActivity
 * @see MainActivity
 */
class MainPresenter  @Inject internal constructor(activity: MainActivity?, appRequest: AppRequest?, userRequest: UserRequest?) : MainBasePresenter(activity, appRequest, userRequest) {
    // ----------------------------------
    // LIFECYCLE
    // ----------------------------------
    // ----------------------------------
    // MENU HANDLING
    // ----------------------------------
    public override fun handleMenu(@IdRes menuId: Int) {
        if (activity == null) return
        when (menuId) {
            R.id.action_update_info -> {
                val privateCircleChooseFragment = PrivateCircleChooseFragment()
                privateCircleChooseFragment.show(activity.supportFragmentManager, PrivateCircleChooseFragment.TAG)
            }
            R.id.action_contact -> {
                val intent = Intent(Intent.ACTION_SENDTO)
                intent.data = Uri.parse("mailto:")
                val addresses = arrayOf(activity.getString(R.string.contact_email))
                intent.putExtra(Intent.EXTRA_EMAIL, addresses)
                if (intent.resolveActivity(activity.packageManager) != null) {
                    activity.startActivity(intent)
                } else {
                    Toast.makeText(activity, R.string.error_no_email, Toast.LENGTH_SHORT).show()
                }
            }
            R.id.action_feedback -> activity.showWebViewForLinkId(Constants.FEEDBACK_ID)
            R.id.action_propose -> Toast.makeText(activity, R.string.error_not_yet_implemented, Toast.LENGTH_SHORT).show()
            R.id.action_how_to -> {
                val howToIntent = Intent(Intent.ACTION_VIEW, Uri.parse(activity.getString(R.string.how_to_url)))
                try {
                    activity.startActivity(howToIntent)
                } catch (ex: Exception) {
                    Toast.makeText(activity, R.string.no_browser_error, Toast.LENGTH_SHORT).show()
                }
            }
            R.id.action_about -> {
                EntourageEvents.logEvent(EntourageEvents.EVENT_MENU_ABOUT)
                val aboutFragment = VoisinageAboutFragment()
                aboutFragment.show(activity.supportFragmentManager, VoisinageAboutFragment.TAG)
            }
            else -> super.handleMenu(menuId)
        }
    }
}